dev/src/write-batch.ts

```diff
@@ -567,25 +614,55 @@ export class WriteBatch {
     );
 
     if (explicitTransaction) {
+      assert(!useBatch, 'Cannot use batch mode with transactions');
       request.transaction = explicitTransaction;
     }
 
-    return this._firestore
-      .request<api.ICommitRequest, api.CommitResponse>('commit', request, tag)
-      .then(resp => {
-        if (request.writes!.length > 0) {
-          const commitTime = Timestamp.fromProto(resp.commitTime!);
-          return resp.writeResults.map(
-            writeResult =>
-              new WriteResult(
-                writeResult.updateTime
-                  ? Timestamp.fromProto(writeResult.updateTime)
-                  : commitTime
-              )
-          );
-        }
-        return [];
-      });
+    if (useBatch) {
+      return this._firestore
+        .request<api.IBatchWriteRequest, api.BatchWriteResponse>(
+          'batchWrite',
+          request as api.IBatchWriteRequest,
+          tag
+        )
+        .then(resp => {
+          const zipped: Array<[
+            api.IWriteResult,
+            google.rpc.IStatus
+          ]> = resp.writeResults.map((result, i) => {
+            return [result, resp.status[i]];
+          });
+          if (request.writes!.length > 0) {
+            return zipped.map(
+              ([writeResult, status]) =>
+                new BatchWriteResult(
+                  writeResult.updateTime
+                    ? Timestamp.fromProto(writeResult.updateTime)
+                    : null,
+                  status.code as Status
```

Comment:
This should be the status with its error message.

Reply:
Added an RpcStatus interface to hold the Status, not sure if that's what you were looking for